tests/test_config.py

```diff
@@ -91,3 +95,26 @@ def test_can_read_toml_rich_types(tmp_path):
 """
     )
     check_configuration_file_reader(config_file)
+
+
+def test_can_read_env_variable(tmp_path):
+    # Check that we can read the "regular" INI .pylintrc file
+    # if it has an environment variable.
+    config_file = tmp_path / "pylintrc"
+    config_file.write_text(
+        """
+[messages control]
+disable = logging-not-lazy,logging-format-interpolation
+jobs = 10
+reports = yes
+"""
+    )
+    os.environ["tmp_path_env"] = str(tmp_path / "pylintrc")
+    options_manager_mix_in = OptionsManagerMixIn("", "${tmp_path_env}")
+    options_manager_mix_in.read_config_file("${tmp_path_env}")
+
```

Comment:
Thank you for the change :) I think we should also check that ``jobs = 10`` or ``reports = yes`` so we make sure that the file was read correctly :)

Reply:
Love the dedication!
I'll see what I can do.

Reply:
I couldn't figure out how to get a provider to set _all_options[opt].
Is there a UML diagram of pylint somewhere?
So I just checked through the parser in load_config_file().

Reply:
Pyliny has pyreverse in it that could generate the uml diagram automatically 😄 Thank you for adding the test and the assert. Looking at the MR it seems you're testing the value in the config file and this could be merged ?

Reply:
I don't know what the  MR is, but yes, the class instance reads the config file and then loads it and I used the parser that is then created to check if the values are the same. So I think the test now tests what  it has to.
Thanks for your feedback btw,  I learned a lot.